jcl/src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -38,8 +38,11 @@ public class Continuation {
 	private final ContinuationScope scope;
 	private final Runnable runnable;
 	private Continuation parent;
-	private boolean started;
-	private boolean finished;
+	/* it's a bit-wise struct of CarrierThread ID and continuation flags(includes started and finished flag)
+	 * low 8 bits are reserved for flags and the rest are the carrier thread ID.
+	 * the state should not be directly accessed from Java, instead using get/set methods(such as isFinished() and setFinished().
+	 */
+	private long state;
```

Comment:
this was declared as volatile while it was in native Continuation.
I guess, it would be ok to do it here too, mostly for documentation purposes, although  not really necessary since ATM there is no java code to directly deal with it